docs/user-guide/styling-your-docs.md

```diff
@@ -153,6 +153,52 @@ Your directory structure should now look like this:
     `theme`. If you want to remove files, or build a theme from scratch, then
     you should review the documentation for building a [custom theme].
 
+#### Overriding Template Blocks
+
+The built-in themes implement many of their parts inside template blocks which
+can be individually overridden in the `main.html` template. Simply create a
+`main.html` template file in your `theme_dir` and define replacement blocks
+within that file. Just make sure that the `main.html` extends `base.html`. For
+example, to alter the title of the MkDocs theme, your replacement `main.html`
+template would contain the following:
+
+```django
+{% extends "base.html" %}
+
+{% block title %}
+<title>Custom title goes here</title>
+{% endblock %}
+```
+
+In the above example, the title block defined in your custom `main.html` file
+will be used in place of the default title block defined in the parent theme.
+You may re-define as many blocks as you desire, as long as those blocks are
+defined in the parent. For example, you could replace the Google Analytics
+script with one for a different service or replace the search feature with your
+own. You will need to consult the parent theme you are using to determine what
+blocks are available to override. The MkDocs and ReadTheDocs themes provide the
+following blocks:
+
+* `site_meta`: Contains meta tags in the document head.
+* `htmltitle`: Contains the page title in the document head.
+* `styles`: Contains the link tags for stylesheets.
+* `scripts`: Contains the JavaScript libraries included in the page.
+* `analytics`: Contains the analytics script.
+* `extrahead`: An empty block in the `<head>` to insert custom tags/scripts/etc.
+* `search_modal`: Contains the search pop-up modal.
```

Comment:
I think the search_modal is the only anomaly here. Since it makes sense in the MkDocs theme, but not the readthedocs theme. Maybe it should be limited more to the search results within the modal (to allow people to replace any text and customise that without revealing the specific details of how the theme is implemented). 

However, I am not sure that is better approach since it still work work well with the read the docs theme, so maybe we shouldn't include it at all. I'm really not sure.